charts/gateway/templates/node-properties-secret.yaml

```diff
@@ -0,0 +1,36 @@
+{{- if and (not .Values.disklessConfig.enabled) (not .Values.disklessConfig.existingSecret.name) }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name:  {{ template "gateway.node.properties" . }}
+  labels:
+    app: {{ template "gateway.name" . }}
+    chart: {{ template "gateway.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    {{- range $key, $val := .Values.additionalLabels }}
+    {{ $key }}: "{{ $val }}"
+    {{- end }}  
+  annotations:
+    description: template for gateway node.properties
+    {{- if  .Values.additionalAnnotations }}
+    {{- range $key, $val := .Values.additionalAnnotations }}
+    {{ $key }}: "{{ $val }}"
+    {{- end }}
+  {{- end }}
+type: Opaque
+stringData:
+  node.properties: |+
+    node.cluster.pass={{ .Values.clusterPassword }}
+    admin.user={{ .Values.management.username }}
+    admin.pass={{ .Values.management.password }}
+    {{- if .Values.database.enabled }}
+    {{- if .Values.database.create }}
+    l7.mysql.connection.url=jdbc:mysql://{{ .Release.Name }}-mysql:3306/{{ .Values.database.name }}
+    {{- else }}
+    l7.mysql.connection.url={{ required "Please set .Values.database.jdbcURL" .Values.database.jdbcURL }}
+    {{- end }}
+    node.db.config.main.user={{ .Values.database.username }}
+    node.db.config.main.pass={{ .Values.database.password }}
```

Comment:
@Gazza7205 when .Values.database.enabled is false ,to handle when derby via node.properties, add 
    {{- else}}
    node.db.type=derby
    node.db.config.main.user={{ .Values.database.username }}

Reply:
updated